databuilder/callback/call_back.py

```diff
@@ -0,0 +1,65 @@
+import abc
+import logging
+import six
+
+from typing import List, Optional  # noqa: F401
+
+LOGGER = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Callback(object):
+    """
+    A callback interface that expected to fire "on_success" if the operation is successful, else "on_failure" if
+    operation failed.
+    """
+
+    @abc.abstractmethod
+    def on_success(self):
+        # type: () -> None
+        """
+        A call back method that will be called when operation is successful
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def on_failure(self):
+        # type: () -> None
+        """
+        A call back method that will be called when operation failed
+        :return: None
+        """
+        pass
```

Comment:
same. should we do raise NotImplementationError to enforce subclass implement the feature?

Reply:
Same as above comment.